### ISSUE TITLE
issues around treatment of rerun.txt - scenarios from same feature on multiple lines

### ISSUE THREAD

**Issue by mcrmfc** [cucumber/cucumber-ruby#865]
Hi,

I am happy to add failing tests and/or pull request for this issue but just want to get a feel for whether this behavior would be considered a genuine bug or whether it's more 'we don't aim to support that'....

When using `parallel_tests` gem with `--group-by scenarios` you can end up with a rerun.txt file that looks something like this:

```
features/test.feature:3
features/test.feature:6
```

If you were not using `parallel_tests` it would look like this:

```
features/test.feature:3:6
```

There was a [change](https://github.com/cucumber/cucumber/commit/30483b9778b9339ff906872d982add386ec6f554) in [v1.3.19](https://github.com/cucumber/cucumber/compare/v1.3.18...v1.3.19) which caused the behaviour of cucumber when encountering a rerun file in the first format to change.

It now only runs the first scenario in this case.

Note: in versions prior to `1.3.19` although both scenarios were run there was another bug that meant when using junit reports that the first result was overwritten by the second scenario run.

The main thing I want to check is whether the core team think rerun files in the format above should be handled...an alternative would be to get `parallel_tests` to write out a rerun file in the second format (that's the current hack I am doing myself in the project I am working on).

I have a [repo](https://github.com/mcrmfc/cucumber-rerun-bug) with a minimal project to replicate this issue if that's of any use.

Thoughts appreciated, cheers.

**Comment by mattwynne** [comment]
I think Cucumber should be able to understand both formats, so yes, a PR would be great.

Can you have a go at factoring out a class to model the Rerun file where we can isolate all this behaviour with specs?

**Comment by mcrmfc** [comment]
Thanks Matt, no worries I will try and look at that.

That change in 1.3.19 actually appears to break cucumber for any rerun.txt with multiple lines.

**Comment by mattwynne** [comment]
Thanks!

It's helpful if you can remember to close and issue if you replace it with a PR - one less bit of clutter in the list for us to deal with :smile:

**Comment by brasmusson** [comment]
Or in the description of the PR use a [keyword for closing issues](https://help.github.com/articles/closing-issues-via-commit-messages/#keywords-for-closing-issues) for instance `"closes #865"` (backticked here so it is not interpreted as GFM in this example), then the issue will be closed automatically when the PR is megerd.

**Comment by mattwynne** [comment]
Yeah that’s the ninja move.

**Comment by Vanuan** [comment]
@mattwynne can we have this released in 1.3.21?

**Comment by mattwynne** [comment]
@Vanuan I'd prefer not to do any more maintenance on 1.3 - can you explain why you haven't upgraded?
